### DISCUSSION TITLE
struct linux_rtc_time promised, struct tm provided, is this ok?

### DISCUSSION THREAD

**Discussion by matthewnourse** [util-linux/util-linux#2104] [Q&A] [answered]
In `sys-utils/hwclock-rtc.c` :
`# define RTC_RD_TIME	_IOR('p', 0x09, struct linux_rtc_time)`

...but in `do_rtc_read_ioctl(int rtc_fd, struct tm *tm)` :
`rc = ioctl(rtc_fd, RTC_RD_TIME, tm);`

On my system the fields in `linux_rtc_time` appear in the same order as `struct tm` so all is well I guess...but... 
...is this guaranteed to be the case everywhere & into the future?
...and is it ok that `tm.tm_gmtoff` and `tm.tm_zone` are uninitialized?

If not & there's a potential problem...would you like a pull request that passes a `linux_rtc_time` to `ioctl` and then copies the results to the `tm` (along the lines of what's already done for sparc)?

**Comment by karelzak** [comment, marked as answer]
just for the record, it should be fixed by commit a423fe19f6edec437c503ed16a718d25add96b72
